runChecks.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/python
+
+"""
+Replacement for test-math-dependencies target in Makefile.
+
+"""
+
+from __future__ import print_function
+import os
+import os.path
+import sys
+import re
+import glob
+
+winsfx = ".exe"
+testsfx = "_test.cpp"
+def files_in_folder(folder):
+    files = []
+    for f in glob.glob(folder):
+        if os.path.isdir(f):
+            files += files_in_folder(f+os.sep+"**")
+        else:
+            files.append(f)
+    return files
+def grep_patterns(type, folder, patterns_and_messages, exclude_filters = []):
+    """Checks the files in the provided folder for matches
+    with any of the patterns. It returns an array of
+    messages and the provided type with
+    the line number. This check ignores comments.
+    """
+    errors = []
+    folder.replace("/", os.sep)
+    exclude_files = []
+    for excl in exclude_filters:
+        exclude_files += files_in_folder(excl)
+    files = files_in_folder(folder+os.sep+"**")
+    files = [x for x in files if x not in exclude_files]
+    for f in files:
```

Comment:
I would use `f` --> `filepath` (`f` is usually filehandle from the `open` function)